pkgs/os-specific/linux/kernel/mainline.nix

```diff
@@ -6,8 +6,14 @@ let
   inherit (thisKernel) version;
 
   src =
+    # -next kernels are a special case because they don't have tarballs on the CDN
+    if lib.hasPrefix "next-" branch
+      then fetchzip {
+        url = "https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/snapshot/linux-next-${branch}.tar.gz";
```

Comment:
```suggestion
        url = "https://git.kernel.org/next/linux-next/t/linux-next-${branch}.tar.gz";
```

Reply:
This doesn't work for me; I get a lump of HTML.

https://git.kernel.org/next/linux-next/t/linux-next-20231107.tar.gz

Reply:
the correct URL is https://git.kernel.org/next/linux-next/t/linux-linux-next-20231107.tar.gz